### PR TITLE
Add preview Lambda Annotation blueprint

### PR DESCRIPTION
**Pull request by normj**
*Description of changes:*
Add blueprint using the new preview Lambda Annotations framework.

Decisions made when creating the blueprint:
*  Lambda project uses `Microsoft.NET.Sdk.Web` instead of `Microsoft.NET.Sdk` so if using dependency injection libraries it won't have to be included in the deployment bundle. Because of this I manually set the `OutputType` to **Library**.
* Didn't wrap the logic in the Functions.cs in a service because I want to make it easy to delete all of the sample code so it can easily be used as a starting point for real projects
* Enabled `ImplicitUsings` in both the Lambda and test project

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Functions.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        /// PackageType is currently required to be set to LambdaPackageType.Image till the upcoming .NET 6 managed
+        /// runtime is available. Once the .NET 6 managed runtime is available PackageType will be optional and will
+        /// default to Zip.
+        /// </summary>
+        /// <param name="x"></param>
+        /// <param name="y"></param>
+        /// <returns>x minus y</returns>
+        [LambdaFunction(PackageType = LambdaPackageType.Image)]
+        [HttpApi(Amazon.Lambda.Annotations.HttpMethod.Get, "/multiply/{x}/{y}")]
+        public int Times(int x, int y, ILambdaContext context)
```
Comment (ganeshnj) [inline Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Functions.cs:90]:
```suggestion
        public int Multiply(int x, int y, ILambdaContext context)
```

File: Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Functions.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        /// PackageType is currently required to be set to LambdaPackageType.Image till the upcoming .NET 6 managed
+        /// runtime is available. Once the .NET 6 managed runtime is available PackageType will be optional and will
+        /// default to Zip.
+        /// </summary>
+        /// <param name="x"></param>
+        /// <param name="y"></param>
+        /// <returns>x minus y</returns>
+        [LambdaFunction(PackageType = LambdaPackageType.Image)]
+        [HttpApi(Amazon.Lambda.Annotations.HttpMethod.Get, "/substract/{x}/{y}")]
+        public int Minus(int x, int y, ILambdaContext context)
```
Comment (ganeshnj) [inline Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Functions.cs:72]:
```suggestion
        public int Substract(int x, int y, ILambdaContext context)
```

File: Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Readme.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
+# .NET Lambda Annotations Framework (Preview)
+
+The Lambda Annotations framework is currently in **preview**. Development for this framework is being done on the [aws/aws-lambda-dotnet](https://github.com/aws/aws-lambda-dotnet) repository. We would appreciate any feedback or feature suggestions on this library. 
+
+Lambda Annotations is a programming model for writing .NET Lambda function. At a high level the programming model allows
+idiomatic .NET coding patterns and uses [C# Source Generators](https://docs.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/source-generators-overview) to bridge the gap between the Lambda programming model to the more .NET idiomatic programming model.
+
+For example here is a simplistic example of a .NET Lambda function that acts like a calculator plus method using the normal 
+Lambda programming model. It respondes to an API Gateway REST API, pulls the operands from the resource paths, does the
```
Comment (ganeshnj) [inline Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Readme.md:9]:
```suggestion
Lambda programming model. It responds to an API Gateway REST API, pulls the operands from the resource paths, does the 
```

File: Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Readme.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+
+* LambdaFunction
+    * Placed on a method. Indicates this method should be exposed as a Lambda function.
+* LambdaStartup
+    * Placed on a class. Indicates this type should be used as the startup class and is used to configure the dependency injection and middleware. There can only be one class in a Lambda project with this attribute.
+
+### Event Attributes    
+
+Event attributes configuring the source generator for the type of event to expect and setup the event source in the CloudFormation temlate. If an event attribute is not set the
```
Comment (ganeshnj) [inline Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Readme.md:135]:
```suggestion
Event attributes configuring the source generator for the type of event to expect and setup the event source in the CloudFormation template. If an event attribute is not set the
```

**Review by ganeshnj — APPROVED** [review]

File: Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Functions.cs (hunk 1 above)
Comment (normj) [reply Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Functions.cs:90]: Fixed

File: Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Functions.cs (hunk 2 above)
Comment (normj) [reply Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Functions.cs:72]: Fixed

File: Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Readme.md (hunk 3 above)
Comment (normj) [reply Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Readme.md:9]: Fixed also in the README.md for the NuGet package

File: Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Readme.md (hunk 4 above)
Comment (normj) [reply Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Readme.md:135]: Fixed also in the README.md for the NuGet package

**Review by ganeshnj — APPROVED** [review]

File: Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Functions.cs [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        /// <summary>
+        /// Perform x * y.
+        ///
+        /// PackageType is currently required to be set to LambdaPackageType.Image till the upcoming .NET 6 managed
+        /// runtime is available. Once the .NET 6 managed runtime is available PackageType will be optional and will
+        /// default to Zip.
+        /// </summary>
+        /// <param name="x"></param>
+        /// <param name="y"></param>
+        /// <returns>x minus y</returns>
```
Comment (boblodgett) [inline Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Functions.cs:87]: x times y or something similar

File: Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Functions.cs [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+        /// <summary>
+        /// Perform x / y.
+        ///
+        /// PackageType is currently required to be set to LambdaPackageType.Image till the upcoming .NET 6 managed
+        /// runtime is available. Once the .NET 6 managed runtime is available PackageType will be optional and will
+        /// default to Zip.
+        /// </summary>
+        /// <param name="x"></param>
+        /// <param name="y"></param>
+        /// <returns>x minus y</returns>
```
Comment (boblodgett) [inline Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Functions.cs:105]: x divided by y or something similar

File: Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Readme.md [hunk 7]
```diff
@@ -0,0 +1,154 @@
+# .NET Lambda Annotations Framework (Preview)
+
+The Lambda Annotations framework is currently in **preview**. Development for this framework is being done on the [aws/aws-lambda-dotnet](https://github.com/aws/aws-lambda-dotnet) repository. We would appreciate any feedback or feature suggestions on this library. 
+
+Lambda Annotations is a programming model for writing .NET Lambda function. At a high level the programming model allows
+idiomatic .NET coding patterns and uses [C# Source Generators](https://docs.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/source-generators-overview) to bridge the gap between the Lambda programming model to the more .NET idiomatic programming model.
+
+For example here is a simplistic example of a .NET Lambda function that acts like a calculator plus method using the normal
```
Comment (boblodgett) [inline Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Readme.md:8]: Wording: `For example here is a simplistic example` => `Here is a simplistic example`

File: Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Readme.md [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+
+* RestApi
+    * Configures the Lambda function to be called from an API Gateway REST API. The HTTP method and resource path are required to be set on the attribute.
+* HttpApi
+    * Configures the Lambda function to be called from an API Gateway HTTP API. The HTTP method, HTTP API payload version and resource path are required to be set on the attribute.
+
+### Parameter Attributes
+
+* FromHeader
+    * Map method parameter to HTTP header value
```
Comment (boblodgett) [inline Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Readme.md:146]: nit: ending `.` missing on 4 out of 5 descriptions.

File: Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Functions.cs (hunk 5 above)
Comment (normj) [reply Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Functions.cs:87]: Fixed

File: Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Functions.cs (hunk 6 above)
Comment (normj) [reply Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Functions.cs:105]: Fixed

File: Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Readme.md (hunk 7 above)
Comment (normj) [reply Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Readme.md:8]: Fixed

File: Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Readme.md (hunk 8 above)
Comment (normj) [reply Blueprints/BlueprintDefinitions/vs2022/AnnotationsServerless-Image/template/src/BlueprintBaseName.1/Readme.md:146]: Fixed

**Review by boblodgett — APPROVED** [review]

**Comment by normj** [comment]
Blueprint has shipped.

**Pull request closed without merging**